recipes/mysql.rb

```diff
@@ -0,0 +1,62 @@
+#
+# Cookbook:: osl-keepalived
+# Recipe:: mysql
+#
+# Copyright:: 2018, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+node.default['osl-keepalived']['virtual_ipv4'] = %w(140.211.9.47/24)
+node.default['osl-keepalived']['virtual_ipv6'] = %w()
+
+node.default['osl-keepalived']['master'] = {
+  'mysql3.osuosl.org' => true,
+  'mysql4.osuosl.org' => false
+}
+
+node.default['osl-keepalived']['priority'] = {
+  'mysql3.osuosl.org' => 200,
+  'mysql4.osuosl.org' => 100
+}
+
+include_recipe 'osl-keepalived::default'
+
+secrets = data_bag_item('osl_keepalived', 'mysql')
+
+keepalived_vrrp_instance 'mysql-ipv4' do
+  master node['osl-keepalived']['master'][node['fqdn']]
+  interface node['network']['default_interface']
+  virtual_router_id 1
+  priority node['osl-keepalived']['priority'][node['fqdn']]
+  authentication auth_type: 'PASS', auth_pass: secrets['auth_pass']
+  virtual_ipaddress node['osl-keepalived']['virtual_ipv4']
+  not_if { node['osl-keepalived']['virtual_ipv4'].empty? }
+end
+
+keepalived_vrrp_instance 'mysql-ipv6' do
+  master node['osl-keepalived']['master'][node['fqdn']]
+  interface node['network']['default_interface']
+  virtual_router_id 2
+  priority node['osl-keepalived']['priority'][node['fqdn']]
+  # Authentication isn't actually used with IPv6 (see https://tools.ietf.org/html/rfc5798#section-9)
+  # however, keepalived cookbook requires this to be defined
+  authentication auth_type: 'PASS', auth_pass: secrets['auth_pass']
+  virtual_ipaddress node['osl-keepalived']['virtual_ipv6']
+  not_if { node['osl-keepalived']['virtual_ipv6'].empty? }
+end
```

Comment:
I don't really see us using IPv6 for the mysql load balancer anytime soon. Mind removing it from the recipe?